pkg/services/validator/maas.go

```diff
@@ -0,0 +1,661 @@
+
+	validateIDNS, err := prompts.ReadBool("Enable Internal DNS validation", true)
+	if err != nil {
+		return err
+	}
+
+	if !validateIDNS {
+		c.Validator.InternalDNSRules = nil
+		return nil
+	}
+	for i, r := range c.Validator.InternalDNSRules {
+		r := r
+		if err := readMaasInternalDNSRule(c, &r, i, ruleNames); err != nil {
+			return err
+		}
+	}
+
+	addRules := true
+	if c.Validator.InternalDNSRules == nil {
+		c.Validator.InternalDNSRules = make([]vpmaasapi.InternalDNSRule, 0)
+	} else {
+		addRules, err = prompts.ReadBool("Add another Internal DNS rule", false)
+		if err != nil {
+			return err
+		}
+	}
+	if !addRules {
+		return nil
+	}
+	for {
+		if err := readMaasInternalDNSRule(c, &vpmaasapi.InternalDNSRule{}, -1, ruleNames); err != nil {
+			return err
+		}
+		add, err := prompts.ReadBool("Add another Internal DNS rule", false)
+		if err != nil {
+			return err
+		}
+		if !add {
+			break
+		}
+	}
+	return nil
+}
+
+// nolint:dupl
+func readMaasInternalDNSRule(c *components.MaasPluginConfig, r *vpmaasapi.InternalDNSRule, idx int, ruleNames *[]string) error {
+	if r == nil {
+		r = &vpmaasapi.InternalDNSRule{
+			MaasDomain:   "",
+			DNSResources: []vpmaasapi.DNSResource{},
+		}
+	}
+
+	err := initMaasRule(r, "Internal DNS", ruleNames)
+	if err != nil {
+		return err
+	}
+
+	if r.MaasDomain == "" {
+		domain, err := prompts.ReadText("MAAS Domain", "maas.io", false, -1)
+		if err != nil {
+			return err
+		}
+		r.MaasDomain = domain
+	}
+
+	addResources := true
+
+	for addResources {
+		resource, err := readMaasDNSResource()
+		if err != nil {
+			return err
+		}
+
+		r.DNSResources = append(r.DNSResources, resource)
+		addResources, err = prompts.ReadBool("Add another DNS resource", false)
+		if err != nil {
+			return err
+		}
+	}
+	if idx == -1 {
+		c.Validator.InternalDNSRules = append(c.Validator.InternalDNSRules, *r)
+	} else {
+		c.Validator.InternalDNSRules[idx] = *r
+	}
+	return nil
+}
+
+// nolint:dupl
+func readMaasDNSResource() (vpmaasapi.DNSResource, error) {
+	res := vpmaasapi.DNSResource{}
+
+	fqdn, err := prompts.ReadDomains("FQDN", "www.maas.io", "must be a valid FQDN", false, 1)
+	if err != nil {
+		return res, err
+	}
+	res.FQDN = fqdn
+
+	addRecord := true
+
+	for addRecord {
+		record, err := readMaasDNSRecord()
+		if err != nil {
+			return res, err
+		}
+
+		res.DNSRecords = append(res.DNSRecords, record)
+		addRecord, err = prompts.ReadBool("Add another DNS record", false)
+		if err != nil {
+			return res, err
+		}
+	}
+	return res, nil
+}
+
+// nolint:dupl
+func readMaasDNSRecord() (vpmaasapi.DNSRecord, error) {
+	log.InfoCLI("Add a DNS Resource Record")
+	rec := vpmaasapi.DNSRecord{}
+
+	ip, err := prompts.ReadIPs("IP", "10.10.10.10", "IP should be a valid IPv4", false, 1)
+	if err != nil {
+		return rec, err
+	}
+	rec.IP = ip
+
+	recType, err := prompts.ReadText("Record type", "A", false, -1)
+	if err != nil {
+		return rec, err
+	}
+	rec.Type = recType
+
+	ttl, err := prompts.ReadText("TTL", "", true, -1)
```

Comment:
Could this be simplified by using ReadInt?

Reply:
TTL is optional but ReadInt doesnt allow for optional input which is why i didnt end up using it